### PR TITLE
Add remainder function

### PR DESCRIPTION
**Pull request by shenh062326**

### CODE REVIEW

**Review by ebyhr — COMMENTED** [review]
Left some initial comments.

File: docs/src/main/sphinx/functions/math.rst [hunk 1]
```diff
@@ -72,6 +72,10 @@ Mathematical functions
 
     Returns the modulus (remainder) of ``n`` divided by ``m``.
 
+.. function:: remainder(n, m) -> [same as input]
```
[the remark this answers is not in the document]
Comment (shenh062326) [reply docs/src/main/sphinx/functions/math.rst:75]: Sorry, I did not find the list-by-topic.rst file.

[the remark this answers is not in the document]
Comment (ebyhr) [reply docs/src/main/sphinx/functions/math.rst:75]: You need to rebase on upstream. 
```bash
git fetch upstream
git rebase upstream/master
```

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java [hunk 2]
```diff
@@ -565,6 +565,85 @@ public void testMod()
...
+        for (double left : doubleLefts) {
+            if (left == 0) {
+                continue;
+            }
+            for (double right : doubleRights) {
+                float fl = (float) left;
+                float fr = (float) right;
+                assertFunction("remainder(REAL '" + fl + "', REAL '" + fr + "')", REAL, (fr - (fl * Math.round(fr / fl))));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (shenh062326) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:645]: Currently decimal is not support.

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java [hunk 3]
```diff
@@ -565,6 +565,85 @@ public void testMod()
...
+
+        for (int left : intLefts) {
+            if (left == 0) {
+                continue;
+            }
+            for (double right : doubleRights) {
+                float fr = (float) right;
+                assertFunction("remainder(" + left + ", REAL '" + fr + "')", REAL, (fr - (left * Math.round(fr / left))));
```
[the remark this answers is not in the document]
Comment (shenh062326) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java]: This tests that the first variable is int and the second variable is REAL.  
The left REAL and right REAL is tested below.

File: docs/src/main/sphinx/functions/math.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (shenh062326) [reply docs/src/main/sphinx/functions/math.rst:75]: Done.

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ebyhr) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:645]: Is there any reason? Also, we can still add negative tests even if it's not supported.

[the remark this answers is not in the document]
Comment (shenh062326) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:645]: Currently DecimalOperators supports modulusScalarFunction, but does not yet support functions similar to Oracle's remainder. If we want to support, we need additional development.

Negative tests already exist, the intLefts, intRights, doubleLefts and doubleRights are contain  negative values.

[the remark this answers is not in the document]
Comment (ebyhr) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:645]: I meant failure case > negative.

[the remark this answers is not in the document]
Comment (shenh062326) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:646]: done

**Comment by colebow** [comment]
:wave: @shenh062326 - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by colebow** [comment]
Closing this one out due to inactivity, but please reopen if you would like to pick this back up.

**Pull request closed without merging**